src/server/pfs/server/driver.go

```diff
@@ -1486,10 +1480,59 @@ func (d *driver) squashCommitSet(txnCtx *txncontext.TransactionContext, commitse
 		}
 	}
 
-	// 5) notify PPS that this commitset has been squashed so it can clean up any
+	return nil
+}
+
+// dropCommitSet is only implemented for commits with no children, so if any
+// commits in the commitSet have children the operation will fail.
+func (d *driver) dropCommitSet(txnCtx *txncontext.TransactionContext, commitset *pfs.CommitSet) error {
+	// Look up the commits in the CommitSet
+	commitInfos, err := d.inspectCommitSetImmediate(txnCtx, commitset)
+	if err != nil {
+		return err
+	}
+
+	for _, ci := range commitInfos {
+		if len(ci.ChildCommits) != 0 {
```

Comment:
nit, I usually prefer > for checking against an empty slice since they cannot have negative lengths.